src/components/Card/Card.tsx

```diff
@@ -32,19 +36,40 @@ const Card = (props: CardProps): JSX.Element => {
         }
     };
 
+    const name = columns.find(col => col.key === 'name')?.value;
+    const displayName = columns.find(col => col.key === 'displayName')?.value;
+    const userLocation = columns.find(col => col.key === 'location')?.value;
+    const avatarUrl = columns.find(col => col.key === 'avatarUrl')?.value;
+
     return (
         <CardContainer
             data-testid={`cardContainer-${id}`}
             hasNavigation={hasNavigation}
             onClick={handleOnClick}
+            isUser={isUser}
+            isLeader={isLeader}
         >
-            {
-                columns.map(({key: columnKey, value}, index) => (
-                    <p data-testid='card-column' key={`${columnKey}-${index}`}>
-                        <strong>{columnKey}</strong>&nbsp;{value}
-                    </p>
-                ))
-            }
+            {isLeader && <CardLeader data-testid='card-leader'>Team Leader</CardLeader>}
```

Comment:
I realize that creating the card template like this limits the use to object types like `UserDataType` and ` TeamType`, but for now, this is what the project requires, if needed in the future to make more generic, this component could be created completely generic and with a different design.